### DISCUSSION TITLE
Error when build in Nextjs 12.1.6

### DISCUSSION THREAD

**Discussion by cungns** [WalletConnect/walletconnect-monorepo#2962] [Developer Support]
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/61767208/a78b1ccd-d418-4908-9720-6b15590b7083)
I'm using wallet connect 2 in my project and when I run "npm run build" have an error
./node_modules/viem/node_modules/abitype/dist/types/human-readable/formatAbi.d.ts:18:35
Type error: Type parameter declaration expected.

export declare function formatAbi<const TAbi extends Abi | readonly unknown[]>(abi: TAbi): FormatAbi;

**Comment by laverix** [comment]
same here

**Reply by cungns** [reply]
Have you resolved it?

**Reply by laverix** [reply]
I can't use new typescript versions in my app, so I changed viem back to 0.3.50 🤷‍♂️

**Reply by crypblizz8** [reply]
Hey there, this seems like a viem, not a WalletConenct issue? If it is the later, can you elaborate more on the situation?
